src/mcu.h

```diff
@@ -186,43 +201,113 @@ struct mcu_t {
     uint8_t interrupt_pending[INTERRUPT_SOURCE_MAX];
     uint8_t trapa_pending[16];
     uint64_t cycles;
+
+    uint8_t rom1[ROM1_SIZE];
+    uint8_t rom2[ROM2_SIZE];
+    uint8_t ram[RAM_SIZE];
+    uint8_t sram[SRAM_SIZE];
+    uint8_t nvram[NVRAM_SIZE];
+    uint8_t cardram[CARDRAM_SIZE];
+
+    uint8_t dev_register[0x80];
+
+    uint16_t ad_val[4];
+    uint8_t ad_nibble;
+    uint8_t sw_pos;
+    uint8_t io_sd;
+
+    submcu_t* sm;
+    pcm_t* pcm;
+    mcu_timer_t* timer;
+    lcd_t* lcd;
+
+    uint32_t uart_write_ptr;
+    uint32_t uart_read_ptr;
+    uint8_t uart_buffer[uart_buffer_size];
+
+    uint8_t uart_rx_byte;
+    uint64_t uart_rx_delay;
+    uint64_t uart_tx_delay;
+
+    int romset;
+
+    int mcu_mk1; // 0 - SC-55mkII, SC-55ST. 1 - SC-55, CM-300/SCC-1
+    int mcu_cm300; // 0 - SC-55, 1 - CM-300/SCC-1
+    int mcu_st; // 0 - SC-55mk2, 1 - SC-55ST
+    int mcu_jv880; // 0 - SC-55, 1 - JV880
+    int mcu_scb55; // 0 - sub mcu (e.g SC-55mk2), 1 - no sub mcu (e.g SCB-55)
+
+    int rom2_mask;
+
+    int ga_int[8];
+    int ga_int_enable;
+    int ga_int_trigger;
+    int ga_lcd_counter;
+
+    SDL_atomic_t mcu_button_pressed;
+
+    uint8_t mcu_p0_data;
+    uint8_t mcu_p1_data;
+
+    int adf_rd;
+
+    uint64_t analog_end_time;
+
+    int ssr_rd;
+
+    uint32_t operand_type;
+    uint16_t operand_ea;
+    uint8_t operand_ep;
+    uint8_t operand_size;
+    uint8_t operand_reg;
+    uint8_t operand_status;
+    uint16_t operand_data;
+    uint8_t opcode_extended;
+
+    void* callback_userdata;
+    mcu_sample_callback sample_callback;
+
+    SDL_mutex *work_thread_lock;
```

Comment:
While you're at it, you could take it further and initialise every var. So for example:

```cpp
    uint16_t ad_val[4] = {};
    uint8_t ad_nibble = {};
    uint8_t sw_pos = {};
    uint8_t io_sd = {};

    submcu_t* sm = nullptr;
    pcm_t* pcm = nullptr;
    mcu_timer_t* timer = nullptr;
    lcd_t* LCD = nullptr;
```

 Not initialising can lead to obscure bugs, and I wasted too much time on that, so these days I always initialise everything, it's cheap to do so and eliminated weird bugs that manifest _only sometime_, depending on the concrete memory layout (e.g. obscure bugs that crash your program every 10th run or something because some var you forgot to init is sometimes zero, and sometimes some random number).

Reply:
I agree, but since memset has been used throughout the codebase to clear state I don't want to do anything that would make these types non-trivial. I thought it would be best to submit a PR as more-or-less C-style C++ in case @nukeykt doesn't approve of C++ features, and if he does then these things can be pretty easily cleaned up later.

Reply:
> I agree, but since memset has been used throughout the codebase to clear state 

Right, I actually missed that 😄 Not used to seeing memset much these days 😄